src/main/java/javaslang/collection/Array.java

```diff
@@ -203,6 +203,42 @@ private Array(Object[] back) {
         return Array.ofAll(Iterator.ofAll(array));
     }
 
+    /**
+     * Returns an Array containing {@code n} values of a given Function {@code f}
+     * over a range of integer values from 0 to {@code n - 1}.
+     *
+     * @param <T> Component type of the Array
+     * @param n The number of elements in the Array
+     * @param f The Function computing element values
+     * @return An Array consisting of elements {@code f(0),f(1), ..., f(n - 1)}
+     * @throws NullPointerException if {@code n} or {@code f} are null
+     */
+    public static <T> Array<T> fill(Integer n, Function<Integer, ? extends T> f) {
```

Comment:
* `n` should be `int`, then there's no need to check for null below
* We should  use `Function<? super Integer, ? extends T>` in order to allow functions that operate on `Number`